en/docs/develop/writing-a-custom-local-authenticator.md

```diff
@@ -179,227 +176,269 @@ Let's begin.
+                    SampleLocalAuthenticator sampleLocalAuthenticator = new SampleLocalAuthenticator();
+                    ctxt.getBundleContext().registerService(ApplicationAuthenticator.class.getName(),
+                            sampleLocalAuthenticator, null);
                     if (log.isDebugEnabled()) {
-                        log.info("BasicCustomAuthenticator bundle is activated");
+                        log.debug("SampleLocalAuthenticator bundle is activated");
                     }
                 } catch (Throwable e) {
-                    log.error("BasicCustomAuthenticator bundle activation Failed", e);
+                    log.error("SampleLocalAuthenticator bundle activation failed.", e);
                 }
             }
-        
+
+            @Deactivate
             protected void deactivate(ComponentContext ctxt) {
+
                 if (log.isDebugEnabled()) {
-                    log.info("BasicCustomAuthenticator bundle is deactivated");
+                    log.debug("SampleLocalAuthenticator bundle is deactivated");
                 }
             }
-        
-            protected void unsetRealmService(RealmService realmService) {
-                log.debug("UnSetting the Realm Service");
-                BasicCustomAuthenticatorServiceComponent.realmService = null;
+
+            public static RealmService getRealmService() {
+
+                return realmService;
             }
-        
+
+            @Reference(name = "realm.service",
+                    service = org.wso2.carbon.user.core.service.RealmService.class,
+                    cardinality = ReferenceCardinality.MANDATORY,
+                    policy = ReferencePolicy.DYNAMIC,
+                    unbind = "unsetRealmService")
             protected void setRealmService(RealmService realmService) {
-                log.debug("Setting the Realm Service");
-                BasicCustomAuthenticatorServiceComponent.realmService = realmService;
+
+                if (log.isDebugEnabled()) {
+                    log.debug("Setting the Realm Service");
+                }
+                SampleLocalAuthenticatorServiceComponent.realmService = realmService;
             }
+
+            protected void unsetRealmService(RealmService realmService) {
+
+                if (log.isDebugEnabled()) {
+                    log.debug("UnSetting the Realm Service");
+                }
+                SampleLocalAuthenticatorServiceComponent.realmService = null;
             }
+        }
         ```
 
 4.  Build the project using maven. (Click to see the sample project
-    [here](../../assets/attachments/org.wso2.custom.authenticator.local.zip)).
+    [here](https://github.com/wso2/samples-is/tree/master/authenticators/components/org.wso2.carbon.identity.sample.local.authenticator)).
 
-5.  Copy the .jar file
+5.  Copy and paste the .jar file
```

Comment:
```
5. Copy the .jar file (`org.wso2.custom.authenticator.local-1.0.0.jar `) from the **target** folder in your project directory and copy it to the `<IS_HOME>/repository/components/dropins ` folder.
```

Is this .jar file name correct?

Reply:
`org.wso2.carbon.auth.local.sample-1.0.0.jar`